File: zombie-survivor/python/tests/survivor_builder.py
from zombie_survivor import Survivor


class SurvivorBuilder:
    def __init__(self) -> None:
        self._name = "Bob"
        self._wounds = 0
        self._zombie_kills = 0
        self._equipment: list[str] = []

    def named(self, name: str) -> "SurvivorBuilder":
        self._name = name
        return self

    def with_wounds(self, wounds: int) -> "SurvivorBuilder":
        self._wounds = wounds
        return self

    def with_zombie_kills(self, kills: int) -> "SurvivorBuilder":
        self._zombie_kills = kills
        return self

    def with_equipment(self, *items: str) -> "SurvivorBuilder":
        self._equipment.extend(items)
        return self

    def build(self) -> Survivor:
        survivor = Survivor(self._name)
        for _ in range(self._zombie_kills):
            survivor.kill_zombie()
        for item in self._equipment:
            survivor.equip(item)
        for _ in range(self._wounds):
            survivor.receive_wound()
        return survivor

File: zombie-survivor/typescript/tests/SurvivorBuilder.ts
import { Survivor } from '../src/Survivor.js';

export class SurvivorBuilder {
  private _name = 'Bob';
  private _wounds = 0;
  private _zombieKills = 0;
  private _equipment: string[] = [];

  named(name: string): this { this._name = name; return this; }
  withWounds(wounds: number): this { this._wounds = wounds; return this; }
  withZombieKills(kills: number): this { this._zombieKills = kills; return this; }
  withEquipment(...items: string[]): this {
    this._equipment.push(...items);
    return this;
  }

  build(): Survivor {
    const survivor = new Survivor(this._name);
    for (let i = 0; i < this._zombieKills; i++) survivor.killZombie();
    for (const item of this._equipment) survivor.equip(item);
    for (let i = 0; i < this._wounds; i++) survivor.receiveWound();
    return survivor;
  }
}
